lib/src/lifecycle_module.dart

```diff
@@ -454,14 +461,22 @@ abstract class LifecycleModule extends SimpleModule with Disposable {
             LifecycleState.resuming
           ]);
     }
-    var previousTransition = _transition?.future;
-    _transition = new Completer<Null>();
+
+    Future pendingTransition;
+    if (_transition != null && !_transition.isCompleted) {
+      pendingTransition = _transition.future;
+    }
+
+    final transition = new Completer<Null>();
+    _transition = transition;
     _state = LifecycleState.suspending;
 
-    _suspend(previousTransition)
-        .then(_transition.complete)
-        .catchError(_transition.completeError);
-    return _transition.future;
+    _suspend(pendingTransition)
+        .then(transition.complete)
+        .catchError((error, trace) {
+      transition.completeError(error, trace);
+    });
+    return transition.future;
```

Comment:
Similarly here, you can get around the need for the extra `transition` var if you setup the completion immediately:

```dart
var previousTransition = _transition?.future;
_transition = new Completer<Null>();
_state = LifecycleState.suspending;

_transition.complete(_suspend(previousTransition));
return _transition.future;
```